cli/src/main/java/com/devonfw/tools/ide/util/FilenameUtil.java

```diff
@@ -28,6 +28,12 @@ public static String getExtension(String path) {
       lastSlash = 0;
     }
     int lastDot = path.lastIndexOf('.');
+
+    // workaround for sourceforge urls
+    if (path.contains("sourceforge") && path.endsWith("download")) {
+      return path.substring(lastDot + 1, lastDot + 4);
```

Comment:
I don't quite understand what the numbers 1 and 4 stand for, so I suppose it's magic numbers in this case.

Reply:
The numbers define the range of extraction to obtain the extension of the downloaded file from the last dot.
For example if the `url www.sourceforge.com/testfile.jar/download` is given, then the substring `jar` is extracted.

However, I replaced `lastDot + 4` with `lastSlash` to avoid conflicts, when the extensions are more than 3 characters long.

Reply:
Thank you for your changes.